core/spec/models/mno_enterprise/user_spec.rb

```diff
@@ -373,6 +372,35 @@ def reload_user
       end
     end
 
+    describe '.activate_otp' do
```

Comment:
You shouldn't be describing the implementation but the behavior

Your spec is currently too coupled to the implementation, if you change it as I suggested (eg: `reload`), it will break the spec although the behavior is still correct.

Reply:
I think I kind of get you and have modified it accordingly. Let me know what you think.